src/parser/priest/discipline/modules/features/PurgeTheWicked.tsx

```diff
@@ -46,47 +50,40 @@ class PurgeTheWicked extends Analyzer {
     return this.ptwApplications - this.ptwCasts;
   }
 
-  get bonusDamage() {
-    // This is just a rough guess of how much extra damage you get from the extra PTW's
-    if (this.ptwApplications === 0) {
-      return 0;
-    }
-    const percentExtraPtw = this.extraPTWs / this.ptwApplications;
-    return percentExtraPtw * this.ptwDamage;
-  }
-
-  onDotCast(event) {
+  onDotCast(event: CastEvent) {
     this.ptwCasts += 1;
-    this.lastCastTarget = event.targetID;
+    if (event.targetID) {
+      this.lastCastTarget = event.targetID;
+    }
   }
 
-  onDotApply(event) {
+  onDotApply(event: ApplyDebuffEvent | RefreshDebuffEvent) {
     this.ptwApplications += 1;
 
     if (event.targetID !== this.lastCastTarget) {
       this.ptwCleaveTracker[event.targetID] = 1;
     }
   }
 
-  onDotRemove(event) {
-    delete(this.ptwCleaveTracker[event.targetID]);
+  onDotRemove(event: RemoveDebuffEvent) {
+    delete (this.ptwCleaveTracker[event.targetID]);
   }
 
-  onDotDamage(event) {
+  onDotDamage(event: DamageEvent) {
     if (this.ptwCleaveTracker[event.targetID]) {
       this.ptwCleaveDamage += event.amount + (event.absorbed || 0);
     }
   }
 
-  suggestions(when) {
+  suggestions(when: When) {
     const uptime = this.uptime || 0;
 
     when(uptime).isLessThan(SuggestionThresholds.PURGE_THE_WICKED_UPTIME.minor)
-      .addSuggestion((suggest, actual, recommended) => {
+      .addSuggestion((suggest: SuggestionFactory, actual: number, recommended: number) => {
         return suggest(<span>Your <SpellLink id={this.dotSpell.id} /> uptime can be improved.</span>)
           .icon(this.dotSpell.icon)
           .actual(`${formatPercentage(uptime)}% uptime`)
-          .recommended(`>${Math.round(formatPercentage(recommended))}% is recommended`)
+          .recommended(`>${formatPercentage(recommended)}% is recommended`)
```

Comment:
Was this `Math.round` causing problems?

Reply:
formatPercentage would return a string, so Math.round which takes a number would have been angry. The way to fix it depends on what the end goal is.. but I'm pretty sure in this case the correct answer is to provide the second argument to the format function to trim off the decimal places

Reply:
What he said ^^ formatPercentage was returning a string.

Reply:
Do you know if you want to keep the decimals? If you do `formatPercentage(recommended, 0)` it will look the same as the previous Math.round (I believe)